File: backend/src/services/EmailService.ts
import nodemailer from 'nodemailer';

export class EmailService {
    private transporter: nodemailer.Transporter;

    constructor() {
        this.transporter = nodemailer.createTransport({
            service: 'gmail',
            auth: {
                user: process.env.EMAIL_USER,
                pass: process.env.EMAIL_PASS,
            },
        });
    }

    // Requisito 1: O subordinado recebe um e-mail sempre que uma nova tarefa for atribuída a ele.
    async sendTaskAssignedEmail(to: string, cc: string, subordinateName: string, description: string) {
        try {
            await this.transporter.sendMail({
                from: `"LEVE Investimentos" <${process.env.EMAIL_USER}>`,
                to,
                cc,
                subject: 'LEVE - Nova Tarefa Atribuída',
                text: `Olá ${subordinateName},\n\nUma nova tarefa foi atribuída a você:\n\n"${description}"\n\nAcesse o sistema para mais detalhes.`,
            });
            console.log(`📧 E-mail de Nova Tarefa enviado para: ${to} (CC: ${cc})`);
        } catch (error) {
            console.error("Erro ao enviar e-mail de nova tarefa:", error);
        }
    }

    // Requisito 2: O gestor recebe um e-mail quando a tarefa for finalizada pelo usuário responsável.
    async sendTaskCompletedEmail(to: string, managerName: string, subordinateName: string, description: string) {
        try {
            await this.transporter.sendMail({
                from: `"LEVE Investimentos" <${process.env.EMAIL_USER}>`,
                to,
                subject: 'LEVE - Tarefa Concluída',
                text: `Olá ${managerName},\n\nO colaborador ${subordinateName} concluiu a seguinte tarefa:\n\n"${description}"\n\nAcesse o sistema para validar.`,
            });
            console.log(`📧 E-mail de Tarefa Concluída enviado com sucesso para: ${to}`);
        } catch (error) {
            console.error("Erro ao enviar e-mail de tarefa concluída:", error);
        }
    }

    async sendTaskUpdatedEmail(to: string, cc: string, subordinateName: string, description: string) {
        try {
            await this.transporter.sendMail({
                from: `"LEVE Investimentos" <${process.env.EMAIL_USER}>`,
                to,
                cc,
                subject: 'LEVE - Tarefa Atualizada',
                text: `Olá ${subordinateName},\n\nUma tarefa atribuída a você foi atualizada:\n\n"${description}"\n\nAcesse o sistema para mais detalhes.`,
            });
            console.log(`📧 E-mail de Tarefa Atualizada enviado para: ${to} (CC: ${cc})`);
        } catch (error) {
            console.error("Erro ao enviar e-mail de tarefa atualizada:", error);
        }
    }
}

File: backend_C#/LeveInvestimentos.API/Services/EmailService.cs
using System.Net;
using System.Net.Mail;

namespace LeveInvestimentos.API.Services
{
    public class EmailService
    {
        private readonly IConfiguration _configuration;

        public EmailService(IConfiguration configuration)
        {
            _configuration = configuration;
        }

        private async Task SendEmailAsync(string to, string? cc, string subject, string body)
        {
            try
            {
                var emailUser = _configuration["EmailSettings:User"];
                var emailPass = _configuration["EmailSettings:Password"];

                if (string.IsNullOrEmpty(emailUser) || string.IsNullOrEmpty(emailPass))
                {
                    Console.WriteLine("Aviso: Credenciais de e-mail não configuradas no appsettings.");
                    return;
                }

                using var smtpClient = new SmtpClient("smtp.gmail.com")
                {
                    Port = 587,
                    Credentials = new NetworkCredential(emailUser, emailPass),
                    EnableSsl = true,
                };

                using var mailMessage = new MailMessage
                {
                    From = new MailAddress(emailUser, "LEVE Investimentos"),
                    Subject = subject,
                    Body = body,
                    IsBodyHtml = false,
                };

                mailMessage.To.Add(to);
                if (!string.IsNullOrEmpty(cc))
                {
                    mailMessage.CC.Add(cc);
                }

                await smtpClient.SendMailAsync(mailMessage);
                Console.WriteLine($"📧 E-mail enviado com sucesso para: {to}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Erro ao enviar e-mail: {ex.Message}");
            }
        }

        public async Task SendTaskAssignedEmailAsync(string to, string cc, string subordinateName, string description)
        {
            var subject = "LEVE - Nova Tarefa Atribuída";
            var body = $"Olá {subordinateName},\n\nUma nova tarefa foi atribuída a você:\n\n\"{description}\"\n\nAcesse o sistema para mais detalhes.";
            await SendEmailAsync(to, cc, subject, body);
        }

        public async Task SendTaskCompletedEmailAsync(string to, string managerName, string subordinateName, string description)
        {
            var subject = "LEVE - Tarefa Concluída";
            var body = $"Olá {managerName},\n\nO colaborador {subordinateName} concluiu a seguinte tarefa:\n\n\"{description}\"\n\nAcesse o sistema para validar.";
            await SendEmailAsync(to, null, subject, body);
        }

        public async Task SendTaskUpdatedEmailAsync(string to, string cc, string subordinateName, string description)
        {
            var subject = "LEVE - Tarefa Atualizada";
            var body = $"Olá {subordinateName},\n\nUma tarefa atribuída a você foi atualizada:\n\n\"{description}\"\n\nAcesse o sistema para mais detalhes.";
            await SendEmailAsync(to, cc, subject, body);
        }
    }
}